### DISCUSSION TITLE
Quarto Date formatting

### DISCUSSION THREAD

**Discussion by nithinmkp** [quarto-dev/quarto-cli#10673] [Q&A]
### Description

I am writing a thesis in quarto. i have university requirements and corresponding latex template. for example in title page i need to give month and year of submission. and also only year in a filed of copyright. I know i can pass date in quarto get year month format for submission. i do this by modifying and using latex partials. What i want to do is use the passed YAML date and use only year part of that in my copyright field. 
 my template looks like this

````{=tex}
$if(has-frontmatter)$
\frontmatter
$endif$

$if(title)$
\thispagestyle{plain}
\begin{center}
\renewcommand{\baselinestretch}{1}\selectfont % Adjust line spacing
{\Large \textbf{$title$}\\}
\vspace*{1.0cm}

$for(by-author)$
$for(by-author.affiliations)$
{\em Thesis submitted to \\ $it.name$\\}
{\em for the award of the degree}\\
\vspace*{0.4cm} {\em of} \\
\vspace*{0.4cm}
{\Large \textbf{Doctor of Philosophy}}\\ % Use degree from YAML
\vspace*{0.4cm}
{\em by}\\
\vspace*{0.4cm}
{\Large \textbf{$by-author.name.literal$}}\\
$endfor$
$endfor$

\vspace*{2cm}
{\em under the guidance of}\\
\vspace*{0.3cm}
{\textbf{Dr. Siddhartha Chattopadhyay}}\\ % Use guide from YAML

$for(by-author)$
$for(by-author.affiliations)$
$if(it.department)$
{\em ($it.department$)} % Print department if available
$endif$
$endfor$
$endfor$

\vspace*{1.6cm}

\begin{figure}[htbp]
  \centering
  \includegraphics[height=4cm,width=3.5cm]{iitlogo.pdf} % Update the path if needed
  %\caption{Institution Logo}
\end{figure}

{\textbf {\uppercase{$for(by-author)$$for(by-author.affiliations)$$it.department$\\
$it.name$$endfor$$endfor$}}\\
\textbf{$date$}}\\
\vspace*{0.25cm} \copyright 2025 $for(by-author)$$by-author.name.literal$$endfor$. All rights reserved.
\end{center}
$endif$
````

what i want to do is in the third last line instead of copyright 2025 which is hardcoded, i want to pass quarto date, something like this 
````{=text}
\vspace*{0.25cm} \copyright year($date$) $for(by-author)$$by-author.name.literal$$endfor$. All rights reserved.
````

minimal qmd file looks like this

````{=qmd}
project:
  type: book

book:
  title: "Consumer Sentiments"
  author: 
    - name: "Nithin M"
      affiliations:
        - name: "Indian Institute of Technology Kharagpur"
          department: "Department of Humanities and Social Sciences"
  chapters:
    - index.qmd
    - summary.qmd
    - references.qmd
  date: "01 June 2026"
  date-format: "MMMM YYYY"

bibliography: references.bib
format:
  pdf:
    documentclass: book

**Comment by mcanouil** [comment]
You need to use a Lua filter to extract the information (metadata) and format it as you need, such as new metadata entries.
Then those new entries can be used in your partials.

**Comment by nithinmkp** [comment]
can you show an example? I actually would love to define new metadata like name of supervisor and the degree too

**Reply by mcanouil** [reply]
You can look at several extension in the listings in Quarto documentation.
Take a look at the documentation on extensions, see <https://quarto.org/docs/extensions/>.

<table>
<tr><th>Quarto</th><th>Lua</th></tr>
<tr><td>

````qmd
---
title: "Quarto Playground"
format: html
date: today
date-format: "YYYY-MM-DD"
filters:
  - get_monthyear.lua
---


Month: {{< meta month >}}
Year: {{< meta year >}}

````

</td><td>

````lua
function Meta(meta)
  local date = pandoc.utils.stringify(meta.date)
  if date then
    local year, month = date:match("(%d%d%d%d)-(%d%d)")
    if year and month then
      meta.year = year
      meta.month = month
    end
  end

  return meta
end

````

</td></tr>
</table>

**Reply by nithinmkp** [reply]
Thanks. lua filter seems difficult at this point

**Comment by nithinmkp** [comment]
one more question here. Should the index.qmd always be inlcuded as the first chapter?
i currently have my Quarto YAML like this

````{=qmd}
project:
  type: book

book:
  title: "Consumer Sentiments"
  author: 
    - name: "Nithin M"
      affiliations:
        - name: "Indian Institute of Technology Kharagpur"
          department: "Department of Humanities and Social Sciences"
  chapters:
    - dedications.qmd
    - index.qmd
    - intro.qmd
    - "Chapter1/chapter_main.qmd"
    - summary.qmd
    - references.qmd
  date: "01 June 2026"
  date-format: "MMMM YYYY"

bibliography: references.bib
format:
  pdf:
    documentclass: scrbook
````

I also have my index.qmd and dedications.qmd like below

index.qmd
````{=qmd}

# Acknowledgment {.unnumbered}
{{< lipsum 2 >}}

\tableofcontents
\listoftables
\listoffigures

# Abbreviations

\let\mainmatter\mainmatterreal
\mainmatter
````

and dedications.qmd

````{=qmd}

```{=latex}
\thispagestyle{empty}%
\vspace*{0.25\textheight}
\begin{center}
{\AlgoFont{14} {\textit {- To my parents -}}}\\
\vspace*{1.0\baselineskip}
{\AlgoFont{16} {\textsf {-To my Laptop-}}}\\
\vspace*{0.5\baselineskip}
\end{center}
\cleardoublepage
```
my idea is to avoid index.qmd altogether and define everything as separate qmd files, like acknowledgment, abbreviations etc. but index.qmd is necessary it seems

**Reply by mcanouil** [reply]
Why do you have `index.qmd`. It's really meaningful only for book website.
`index` means the first page. It does not make sense to have it in the middle.

See the documentation: https://quarto.org/docs/books/book-structure.html

**Comment by nithinmkp** [comment]
I tried it .  Seems not working . This is the error

PS C:\Users\Nithin M\OneDrive\Documents\GitHub\quarto tex experiments\book> quarto render
ERROR: Book contents must include a home page (e.g. index.md)

now my chapters part of YAML looks like this

````{=qmd}
project:
  type: book

book:
  title: "Consumer Sentiments"
  author: 
    - name: "Nithin M"
      affiliations:
        - name: "Indian Institute of Technology Kharagpur"
          department: "Department of Humanities and Social Sciences"
  chapters:
    - dedications.qmd
    - acknowl.qmd
    - intro.qmd
    - "Chapter1/chapter_main.qmd"
    - summary.qmd
    - references.qmd
  date: "01 June 2026"
  date-format: "MMMM YYYY"

bibliography: references.bib
format:
  pdf:
    documentclass: scrbook
````
 and acknwl.qmd

````{=qmd}
# Acknowledgment {.unnumbered}
{{< lipsum 2 >}}

\tableofcontents
\listoftables
\listoffigures

# Abriviations

\let\mainmatter\mainmatterreal
\mainmatter
````

**Reply by mcanouil** [reply]
Then the `index.qmd` is required even for non-html output.
I suggest you rename `dedications.qmd` into `index.qmd` then.

**Comment by nithinmkp** [comment]
I thought of that. but writing a thesis we need many more pages in between. was wondering about a general solution.  Thanks for chipping in always

**Reply by mcanouil** [reply]
Sorry, I don't see the issue to have your first page/chapter be `index.qmd`.

PS: I know exactly what and how it is to write a thesis. There is no need to explain that to me ...

**Reply by nithinmkp** [reply]
Am sorry if It offended you. I did not mean it!! Sorry!!

**Reply by mcanouil** [reply]
It's fine I am not offended but be cautious.
